### PR TITLE
APP-31735 allow for Kubernetes environment variables

### PR DESCRIPTION
**Pull request by mccall114**
To be able to deploy pendo-appengine to GKE, we mostly just need to ensure the environment variables are set in the K8s deployment (`GOOGLE_CLOUD_PROJECT`).

Instead of setting `GAE_*` variables in K8s, I created them in GKE as `K8S_*` and updated appwrap to use a separate struct when running in K8s.

Since K8s will also not be using the appspot.com domain, I've made the domain configurable with an env var and removed the version since we don't have a way to do lookasides (yet).

Example K8s yaml environment variables:
```
        - name: GOOGLE_CLOUD_PROJECT
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: K8S_NODE
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        - name: K8S_POD
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: K8S_SERVICE
          valueFrom:
            fieldRef:
              fieldPath: metadata.labels['app']
        - name: K8S_VERSION
          value: *version
        - name: K8S_ZONE
          valueFrom:
            configMapKeyRef:
              key: K8S_ZONE
              name: jobs-large-config
        - name: K8S_DOMAIN
          valueFrom:
            configMapKeyRef:
              key: K8S_DOMAIN
              name: jobs-large-config
```

---
This change is [Reviewable](https://reviewable.io/reviews/pendo-io/appwrap/93)

### CODE REVIEW

File: appinfok8s.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// ModuleDefaultVersionID not implemented in K8s
+func (ai AppengineInfoK8s) ModuleDefaultVersionID(moduleName string) (string, error) {
+	return "", nil
+}
+
+// NumInstances not implemented in K8s
+func (ai AppengineInfoK8s) NumInstances(moduleName, version string) (int, error) {
+	return 0, nil
```
Comment (jbornemann) [inline appinfok8s.go:59]: If NumInstances means number of pods in the service in this implementation; we can certainly get that information - just not from an environment variable

File: appinfok8s.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+func (ai AppengineInfoK8s) VersionID() string {
+	return os.Getenv("K8S_VERSION")
+}
+
+func (ai AppengineInfoK8s) Zone() string {
+	return os.Getenv("K8S_ZONE")
+}
+
+// ModuleHasTraffic not implemented in K8s
+func (ai AppengineInfoK8s) ModuleHasTraffic(moduleName, moduleVersion string) (bool, error) {
```
Comment (jbornemann) [inline appinfok8s.go:48]: We will have to see what this means - if it means "could receive traffic" as in the service readiness probe is healthy - then that is easy info to get. If it means we are actively receiving traffic, that is more difficult; but possible

File: appinfok8s.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	return false, nil
+}
+
+// ModuleDefaultVersionID not implemented in K8s
+func (ai AppengineInfoK8s) ModuleDefaultVersionID(moduleName string) (string, error) {
```
Comment (jbornemann) [inline appinfok8s.go:53]: Does this map to the version of the deployment perhaps?

File: appinfok8s.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package appwrap
+
+import (
+	"context"
+	"fmt"
+	"os"
+)
+
+type AppengineInfoK8s struct {
```
Comment (jbornemann) [inline appinfok8s.go:9]: If we are implementing an interface; we will want to make sure liskov substitution principle holds true; or else things are bound to break given time. If the interface does not make sense for all implementations (which this at the surface doesn't seem to be the case); it probably makes sense to start looking at a different abstraction for implementations to share.

File: appinfok8s.go (hunk 2 above)
Comment (mccall114) [reply appinfok8s.go:60]: We're using this for the latter, so this would depend on how we do traffic splitting

Comment (jbornemann) [reply appinfok8s.go:60]: Yeah, if we decide to use istio, might have to query the virtual service here for traffic percentage to this version.

**Pull request closed without merging**